src/Nowruz/modules/general/components/iconDropDown/useIconDropDown.tsx

```diff
@@ -22,13 +22,19 @@ export const useIconDropDown = () => {
     await nonPermanentStorage.set({ key: 'identity', value: accountId });
     identities()
       .then((resp) => dispatch(setIdentityList(resp)))
-      .then(() => navigate('/jobs'))
+      .then((resp) => {
+        const current = resp.payload.find((item) => item.id === accountId);
+        const path =
+          current?.type === 'users'
+            ? `profile/users/${(current.meta as UserMeta).username}/view`
```

Comment:
I think I asked to refactor this before, since that PR has been closed then it revert back from the original